docs/website/docs/user-guides/advanced/connecting-a-service.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Connecting to a service
+sidebar_position: 1
+---
+
+This tutorial will show you how you can connect your Go application to a mongodb service.
+
+Building on top of the [Go quickstart guide](../quickstart/go.md), we will extend the application to check if it is connected to a mongodb service.
+
+### Pre-requisites
+1. Install Service Binding Operator on your cluster.  See [installation instruction](https://operatorhub.io/operator/service-binding-operator).
+
+2. Install Percona Server Mongodb operator on your cluster. See [installation instruction](https://operatorhub.io/operator/percona-server-mongodb-operator).
+:::note
+The operator will be installed in a new namespace called "my-percona-server-mongodb-operator" and will be usable from that namespace only.
+:::
+3. Create a mongodb service.
+
+import CreateMongodbService from './_create-mongodb-service.mdx';
+
+<CreateMongodbService/>
+
+### Implement the code logic
+:::note
+If you're already running `odo dev` in a terminal, exit it and start afresh.
+:::
+
+The new code is simple. We obtain the connection information (username, password, and host) from the environment, and use it to connect to the mongodb service and ping it. 
+
+Replace the content of your `main.go` with the following content:
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// Connection URI
+var (
+	USERNAME = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_USER")
+	PASSWORD = os.Getenv("PERCONASERVERMONGODB_MONGODB_USER_ADMIN_PASSWORD")
+	HOST     = os.Getenv("PERCONASERVERMONGODB_HOST")
+	uri      = fmt.Sprintf("mongodb://%s:%s@%s:27017/?maxPoolSize=20&w=majority", USERNAME, PASSWORD, HOST)
+)
+
+func main() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe("0.0.0.0:8080", nil)
+}
+
+func HelloServer(w http.ResponseWriter, r *http.Request) {
+	// Create a new client and connect to the server
+	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
+
+	if err != nil {
+		fmt.Fprintf(w, "failed to connect: %s", err.Error())
+		return
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			fmt.Fprintf(w, "failed to disconnect: %s", err.Error())
+			return
+		}
+	}()
+
+	// Ping the primary
+	if err := client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		fmt.Fprintf(w, "unable to connect to the server: %s", err.Error())
+		return
+	}
+
+	fmt.Fprintf(w, "Successfully connected and pinged.")
+}
+```
+We will be using the mongodb client library, so let's update the go.mod with this dependency by running the following command: 
+```shell
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+### Run the application on the cluster
+Run this application on the cluster with `odo dev`.
```

Comment:
```suggestion
Run the application with `odo dev`.
```

Reply:
I'd rather keep `on the cluster` as is because it makes sense.